src/test/java/ua/edu/internship/interview/service/business/UserSkillsServiceTest.java

```diff
@@ -0,0 +1,180 @@
+
+        userSkillsDocument = new UserSkillsDocument();
+        userSkillsDocument.setUserId("1");
+        userSkillsDocument.setSkills(skillDocuments);
+
+        userSkillsDto = new UserSkillsDto();
+        userSkillsDto.setUserId("1");
+        userSkillsDto.setSkills(skillDTOs);
+    }
+
+    @Test
+    void getUserSkills_shouldReturnUserSkillsDto_whenSkillsExist() {
+        String userId = "1";
+        when(userSkillRepository.findByUserId(userId)).thenReturn(Optional.of(userSkillsDocument));
+        when(userSkillsMapper.toDto(userSkillsDocument)).thenReturn(userSkillsDto);
+
+        UserSkillsDto result = underTest.getUserSkills(userId);
+
+        assertNotNull(result);
+        assertEquals(userId, result.getUserId());
+        assertEquals(2, result.getSkills().size());
+        matchSkillFields(skillDTOs.getFirst(), result.getSkills().getFirst());
+        matchSkillFields(skillDTOs.getLast(), result.getSkills().getLast());
+        verify(userSkillRepository).findByUserId("1");
+    }
+
+    @Test
+    void getUserSkills_shouldThrowNoSuchEntityException_whenUserSkillsNotFound() {
+        String userId = "1";
+        when(userSkillRepository.findByUserId(userId)).thenReturn(Optional.empty());
+
+        assertThrows(NoSuchEntityException.class, () -> underTest.getUserSkills(userId));
+        verify(userSkillRepository).findByUserId(userId);
+    }
+
+    @Test
+    void createUserSkills_shouldCreateUserSkillsDocument_whenUserSkillsDocumentNotExist() {
+        String userId = "1";
+        List<String> skillIds = List.of("1", "2");
+        when(userSkillRepository.findByUserId(userId)).thenReturn(Optional.empty());
+        when(skillRepository.findAllById(skillIds)).thenReturn(skillDocuments);
+        when(userSkillRepository.save(any(UserSkillsDocument.class))).thenReturn(userSkillsDocument);
+        when(userSkillsMapper.toDto(userSkillsDocument)).thenReturn(userSkillsDto);
+
+        UserSkillsDto result = underTest.createUserSkills(userId, skillIds);
+
+        assertNotNull(result);
+        assertEquals("1", result.getUserId());
+        assertEquals(2, result.getSkills().size());
+        assertEquals("Java", result.getSkills().getFirst().getName());
+        assertEquals("Kotlin", result.getSkills().getLast().getName());
+        verify(userSkillRepository).save(userSkillsDocument);
+    }
+
+    @Test
+    void createUserSkills_shouldThrowInvalidInputException_whenUserSkillsExist() {
+        String userId = "1";
+        List<String> skillIds = List.of(userId);
+        when(userSkillRepository.findByUserId(userId)).thenReturn(Optional.of(userSkillsDocument));
+
+        assertThrows(InvalidInputException.class, () -> underTest.createUserSkills(userId, skillIds));
+        verify(userSkillRepository).findByUserId(userId);
+    }
+
+    @Test
+    void updateUserSkills_shouldUpdateUserSkills_whenUserSkillsExist() {
+        String userId = "1";
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.VariableDeclarationUsageDistanceCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Distance between variable 'userId' declaration and its first usage is 4, but allowed 3.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).
<!-- __reviewdog__:ChBjZDE2OGRkZGFlNDMzYTZkEgpjaGVja3N0eWxl -->